### PR TITLE
enable eslint-plugin-jest

### PR DESCRIPTION
**Pull request by MaxBittker**
https://github.com/jest-community/eslint-plugin-jest

this lint ruleset gives us some useful things like:
`no-focused-tests`
`no-identical-title` 
`no-jest-import`
`prefer-to-have-length`
`valid-expect` <- this one found some mistakes in our existing tests

I disabled this one:
`no-disabled-tests`
and enabled a warning for snapshots over 2000 lines, currently triggering only twice:
```
sentry/tests/js/spec/components/actions/__snapshots__/ignore.spec.jsx.snap
     3:1  warning  Expected Jest snapshot to be smaller than 2000 lines but was 5055 lines long  jest/no-large-snapshots
  5060:1  warning  Expected Jest snapshot to be smaller than 2000 lines but was 2246 lines long  jest/no-large-snapshots
```

this should probably live in https://github.com/getsentry/eslint-config-sentry, shouldn't it?

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -27,13 +25,13 @@
...
-    "getsentry"
-  ],
+  "plugins": ["react", "import", "getsentry", "jest"],
   "settings": {
     "import/resolver": "webpack",
     "import/extensions": [".js", ".jsx"]
+  },
+  "rules": {
+    "jest/no-large-snapshots": ["warn", {"maxSize": 2000}],
+    "jest/no-disabled-tests": "off"
```
Comment (billyvg) [inline .eslintrc]: Why is this off?

**Comment by billyvg** [comment]
Yeah this should live in https://github.com/getsentry/eslint-config-sentry

File: .eslintrc (hunk 1 above)
Comment (lynnagara) [reply .eslintrc]: warn would be ok here i think

File: tests/js/spec/components/shareIssue.spec.jsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ describe('ShareIssue', function() {
     expect(wrapper).toMatchSnapshot();
   });
 
-  it('renders when not shared', function() {
+  it('renders when shared ', function() {
```
Comment (billyvg) [inline tests/js/spec/components/shareIssue.spec.jsx]: Was the test description wrong? Also trailing space

**Review by billyvg — APPROVED** [review]

**Pull request merged**